src/main/java/io/neonbee/data/internal/metrics/ConfiguredDataVerticleMetrics.java

```diff
@@ -0,0 +1,160 @@
+            DataVerticleMetrics reportTimingMetric) {
+        this.reportNumberOfRequests = reportNumberOfRequests;
+        this.reportActiveRequestsGauge = reportActiveRequestsGauge;
+        this.reportStatusCounter = reportStatusCounter;
+        this.reportTimingMetric = reportTimingMetric;
+    }
+
+    /**
+     * Configure the {@link DataVerticleMetrics} to report.
+     *
+     * To enable all metrics, you must specify the enabled key with the value ture. Otherwise, no metrics will be
+     * reported. You can select the registry to use by specifying the meterRegistryName key with the name of the
+     * registry associated with the registry in the micrometer options.
+     *
+     * If you specify any of the configuration values "reportNumberOfRequests", "reportActiveRequests",
+     * "reportStatusCounter", "reportTiming", only the values configured as true will be reported. If you do not specify
+     * any of these values, all metrics are reported.
+     *
+     * Full example:
+     *
+     * <pre>
+     * {@code
+     * {
+     *     "enabled" : true,
+     *     "meterRegistryName" : "default",
+     *     "reportNumberOfRequests" : true,
+     *     "reportActiveRequests" : true
+     *     "reportStatusCounter" : true,
+     *     "reportTiming" : true
+     * }
+     * }
+     * </pre>
+     *
+     * @param metricsConfig {@link JsonObject} containing the metrics configuration.
+     * @return configured {@link DataVerticleMetrics}
+     */
+    public static DataVerticleMetrics configureMetricsReporting(JsonObject metricsConfig) {
+        if (metricsConfig == null || !Boolean.TRUE.equals(metricsConfig.getBoolean(ENABLED))) {
+            return DUMMY_IMPL;
+        }
+
+        String meterRegistryName =
+                metricsConfig.getString(METER_REGISTRY_NAME, MicrometerMetricsOptions.DEFAULT_REGISTRY_NAME);
+        MeterRegistry registry = BackendRegistries.getNow(meterRegistryName);
+
+        if (registry == null) {
+            LOGGER.error(
+                    "Micrometer registry hasn't been registered yet or it has been stopped. Metrics will not be sent.");
+            return DUMMY_IMPL;
+        } else {
+            DataVerticleMetrics metricsImpl = new DataVerticleMetricsImpl(registry);
+            return configureDataVericleMetrics(metricsConfig, metricsImpl);
+        }
+    }
+
+    private static DataVerticleMetrics configureDataVericleMetrics(JsonObject metricsConfig,
+            DataVerticleMetrics metricsImpl) {
+
+        int fieldNameSize = metricsConfig.containsKey(METER_REGISTRY_NAME) ? 2 : 1;
+        boolean activateAllMetrics = metricsConfig.size() == fieldNameSize;
+        if (activateAllMetrics) {
+            return metricsImpl;
+        } else {
+            return new ConfiguredDataVerticleMetrics(
+                    Boolean.TRUE.equals(metricsConfig.getBoolean(NUMBER_OF_REQUESTS)) ? metricsImpl : DUMMY_IMPL,
```

Comment:
```suggestion
                    metricsConfig.getBoolean(NUMBER_OF_REQUESTS, false) ? metricsImpl : DUMMY_IMPL,
```

Reply:
With your suggestion the case that the key is present but with a null value would cause an NPE.